splink/profile_data.py

```diff
@@ -231,10 +252,24 @@ def profile_columns(linker, column_expressions=None, top_n=10, bottom_n=10):
             values to display in the respective charts.
     """
 
+    tables = ensure_is_list(table_or_tables)
+
+    tables = db_api.process_input_tables(tables)
+    # TODO: is this sensible?:
+    input_aliases = [f"__splink__profile_data_{idx}" for idx, _ in enumerate(tables)]
```

Comment:
This bit is really just providing a convenience for passing in other objects - pandas frames, spark frames, or a csv path (in duckdb), and making sure that after this point we are dealing consistently with `SplinkDataFrame`s.
`register_multiple_tables` _should_ just be providing a splink alias rather than actually touching the db in cases where we pass in table name (i.e. just wrapping table names as a `SplinkDataFrame`)

Reply:
Right - yeah - got it 

Not for this PR, but I wonder whether the `db_api.register_multiple_tables and _table_registration` should make input aliases optional.

For a use case like this they could just be called __splink__temp_{hash}

Or perhaps it could take a 'label' argument or similar of just 'profile', which would internally be turned into `__splink__profile_{hash}` or something

This operation (I have some dataframes that I want to temporarily use and then ditch) seems sufficiently common to want to avoid having to write this sort of code